### ISSUE TITLE
Unable to create Azure resource lock via Terraform 

### ISSUE THREAD

**Issue by i89825** [hashicorp/terraform-provider-azurerm#14545]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
```
Terraform v1.0.11
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azurerm v2.88.1
```

Hi there,

We started receiving the below error while creating Azure resource lock via terraform. The issue started from Friday (03 Dec 2021), prior to this it was working fine. We upgraded the terraform version to the latest (1.0.11) but still same error:

```
azurerm_management_lock.rglock: Creating...
╷
│ Error: parsing "resource-group-level": parsing scope prefix: unable to find the scope prefix from the value "resource-group-level" with the regex "^((.){1,})/providers/Microsoft.Authorization/locks/(.){1,}"
```

While configuring lock for any Azure resource via Terraform we get this error. Strange thing is that it works fine via azure cli or directly via portal.

### Affected Resource(s)
* `azurerm_management_lock`

### Terraform Configuration Files

```
Main.tf
-------
data "azurerm_resource_group" "example" {
  name     = "testing-terraform-01"
}

resource "azurerm_management_lock" "rglock" {
  name       = "resource-group-level"
  scope      = data.azurerm_resource_group.example.id
  lock_level = "ReadOnly"
  notes      = "This Resource Group is Read-Only"
}
```

### Debug Output
[terraform.txt](https://github.com/hashicorp/terraform-provider-azurerm/files/7677580/terraform.txt)

### Expected Behaviour

Terraform should have create Read-only lock for the Azure resource group.

### Actual Behaviour

We are getting the below error and resource is created with status "tainted".
Attached it the statefile for reference.
[terraform copy.tfstate.txt](https://github.com/hashicorp/terraform-provider-azurerm/files/7677604/terraform.copy.tfstate.txt)

### Steps to Reproduce

Use the terraform module for Azure resource lock to create lock for any Azure resource 

1. `terraform apply`

### Important Factoids

The issue exists in every Azure region 

### References

https://github.com/hashicorp/terraform/issues/30108

* #0000

**Comment by pszypowicz** [comment]
Also possible to reproduce with 

```
resource "azurerm_resource_group" "test" {
  name     = "test_rg"
  location = "West Europe"
}

resource "azurerm_public_ip" "test_pip" {
  name                = "test_pip"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  allocation_method   = "Static"
  sku                 = "Standard"
}

resource "azurerm_management_lock" "rglock" {
  name       = "resource-group-level"
  scope      = azurerm_resource_group.test.id
  lock_level = "ReadOnly"
  notes      = "This Resource Group is Read-Only"
}

resource "azurerm_management_lock" "resource_lock" {
  name       = "resource-level"
  scope      = azurerm_public_ip.test_pip.id
  lock_level = "ReadOnly"
  notes      = "This Resource Group is Read-Only"
}
```

and the error is

```
azurerm_management_lock.resource_lock: Creating...
╷
│ Error: parsing "resource-group-level": parsing scope prefix: unable to find the scope prefix from the value "resource-group-level" with the regex "^((.){1,})/providers/Microsoft.Authorization/locks/(.){1,}"
│ 
│   with azurerm_management_lock.rglock,
│   on test.tf line 18, in resource "azurerm_management_lock" "rglock":
│   18: resource "azurerm_management_lock" "rglock" {
│ 
╵
╷
│ Error: parsing "resource-level": parsing scope prefix: unable to find the scope prefix from the value "resource-level" with the regex "^((.){1,})/providers/Microsoft.Authorization/locks/(.){1,}"
│ 
│   with azurerm_management_lock.resource_lock,
│   on test.tf line 25, in resource "azurerm_management_lock" "resource_lock":
│   25: resource "azurerm_management_lock" "resource_lock" {
│ 
```

**Comment by wernerfred** [comment]
Related #14536 and #14495

**Comment by i89825** [comment]
Issue fixed with Latest version of azurerm plugin (2.89.0)
